### DISCUSSION TITLE
Access `document` instance in onResolution hook

### DISCUSSION THREAD

**Discussion by salmanm** [mercurius-js/mercurius#577] [Q&A] [answered]
In `preExecution` hook, we get `schema`, `document` and `context` params. But in `onResolution`, we don't get `document`. 

It makes sense in `preExecution`, coz it allows to modify it, but it would be helpful to get access to it in the `onResolution` too. It should be useful to perform some actions based on it. For example, I'm writing audit logs based on query name.

Thoughts?

**Comment by mcollina** [comment]
Seems like a good feature to add.

**Comment by salmanm** [comment, marked as answer]
If we look at the function signature, `document` should be before `context` in `onResolution`. But then that would be a breaking change!

##### preExecution
```js
fastify.graphql.addHook('preExecution', async (schema, document, context) => {
```

##### onResolution
```js
fastify.graphql.addHook('onResolution', async (execution, context) => {
```

For now, I'll just use [this feature](https://github.com/mercurius-js/mercurius/blob/master/test/hooks.js#L428). Didn't know it exists. 😄

**Reply by mcollina** [reply]
Feel free to send the PR, I would land it when we bump the major.

**Reply by dillonius01** [reply]
Hi @mcollina , are you still open to a breaking change here for adding `document` to the `onResolution` handler? I would be happy to work on a PR but wanted to check with you first. We're currently hacking a workaround where we leverage a `preExecution` hook to append metadata from `document` onto `context` (use case is to drive some metrics reporting). Please let me know, and thanks!
